File: Leetcode_682.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        b = []
        for i in range(len(operations)):
            if operations[i] == '+':
                b.append(b[-1] + b[-2])
            elif operations[i] == 'D':
                b.append(2 * b[-1])
            elif operations[i] == 'C':
                b.pop()
            else:
                b.append(int(operations[i]))
        return sum(b)
        

File: Leetcode_682.java
class Solution {
    public int calPoints(String[] operations) {
        List<Integer> b = new ArrayList<>();

        for (int i = 0; i < operations.length; i++) {
            String op = operations[i];

            if (op.equals("+")) {
                int sum = b.get(b.size() - 1) + b.get(b.size() - 2);
                b.add(sum);
            } else if (op.equals("D")) {
                b.add(2 * b.get(b.size() - 1));
            } else if (op.equals("C")) {
                b.remove(b.size() - 1);
            } else {
                b.add(Integer.parseInt(op));
            }
        }

        int total = 0;
        for (int num : b) {
            total += num;
        }

        return total;
    }
}
